src/librustc_allocator/expand.rs

```diff
@@ -97,7 +97,7 @@ impl<'a> Folder for ExpandAllocatorDirectives<'a> {
         ]);
         let mut items = vec![
             f.cx.item_extern_crate(f.span, f.alloc),
-            f.cx.item_use_simple(f.span, Visibility::Inherited, super_path),
+            f.cx.item_use_simple(f.span, dummy_spanned(VisibilityKind::Inherited), super_path),
```

Comment:
Dummy spans should not be used, unless completely unavoidable.
In most cases during expansion spans for generated code can be inherited from something. In this case `item.span` is used for the path and for the extern crate item, it can be used for visibility as well.

Reply:
>Dummy spans should not be used, unless completely unavoidable.

It's probably not so dire for visibilities because they don't have hygiene, it's still desirable to use some valid spans when possible.

Reply:
@petrochenkov Thank you for your advice!

Is it acceptable to use 'empty' span like the following, rather than `item.span`? I think it makes easier to handle comments in rustfmt and reflects the syntactic structure of source code more accurately (as `Inherited` visibility has no keyword).

```rust
Span { hi: item.span.lo(), ..item.span }
```

Reply:
Yes, that would be even better.
(I was concerned about `item.span.lo` pointing to the location before item attributes, but it looks like this is not the case.)